### PR TITLE
Optionally compress files prior to uploading them in SingularityS3Uploader.

### PR DESCRIPTION
**Pull request by baconmania**

### CODE REVIEW

File: SingularityS3Uploader/src/main/java/com/hubspot/singularity/s3uploader/SingularityUploader.java [hunk 1]
```diff
@@ -255,7 +257,38 @@ private AtomicInteger handleFile(Path path, boolean isFinished, List<Path> toUpl
 
     found.incrementAndGet();
 
-    toUpload.add(path);
+    if (uploadMetadata.isCompressBeforeUpload() && !path.toString().endsWith(".gz")) {
```
Comment (ssalinas) [inline SingularityS3Uploader/src/main/java/com/hubspot/singularity/s3uploader/SingularityUploader.java:260]: if we are checking for 'isAlreadyGzipped' here, should we also be checking that in the file glob section? E.g. if something else already zipped it, it should still match the toUpload glob as well

Comment (ssalinas) [reply SingularityS3Uploader/src/main/java/com/hubspot/singularity/s3uploader/SingularityUploader.java:260]: nevermind, stared down the section above a bit more, we are catching those other cases

**Comment by ssalinas** [comment]
🚢

**Comment by tpetr** [comment]
:ship:

**Pull request merged**